### PR TITLE
Update finalfusion dependency.

### PR DESCRIPTION
**Pull request by sebpuetz**
Incorporate changes from finalfusion and temporarily point to
github.

Uses `is_some` as suggested by @danieldk in #75

### CODE REVIEW

File: src/vocab.rs [hunk 1]
```diff
@@ -35,22 +35,29 @@ impl PyVocab {
         })
     }
 
-    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, usize)>>> {
+    fn ngram_indices(&self, word: &str) -> PyResult<PyObject> {
```
Comment (sebpuetz) [inline src/vocab.rs]: Clippy complained about too complex return type with `PyResult<Option<Vec<(String, Option<usize>)>>>`. Returning `PyObject` should be simpler.

Comment (danieldk) [reply src/vocab.rs]: I don't like it, since it makes it harder to just read the signatures and understand what a function does. IMO, Clippy is too overzealous in this case, since the type is still readable. Better to add an allow to silence Clippy on this one.

File: src/vocab.rs [hunk 2]
```diff
@@ -75,4 +82,13 @@ impl PySequenceProtocol for PyVocab {
             Ok(words[idx as usize].clone())
         }
     }
+
+    fn __contains__(&self, word: String) -> PyResult<bool> {
```
Comment (danieldk) [inline src/vocab.rs]: Seems to be an unrelated change of another PR?

**Review by danieldk — CHANGES_REQUESTED** [review]

File: src/vocab.rs (hunk 2 above)
Comment (sebpuetz) [reply src/vocab.rs]: Forgot to rebase this on master after merging the `contains` PR, will do that now.

**Comment by sebpuetz** [comment]
For some reason I can't comment directly in context on this:

> I don't like it, since it makes it harder to just read the signatures and understand what a function does. IMO, Clippy is too overzealous in this case, since the type is still readable. Better to add an allow to silence Clippy on this one.

There's still an upside to this: It allows for polymorphic return types. E.g. in case of `WordIndex::Word(idx)` we could return `idx` as is and it'd appear in python as an `int`. As of now, we have to differentiate between `WordIndex::Subword` and `WordIndex::Word` through a call to `__contains__`.

_edit:_ just realized I was referring to the wrong method here. So not even related to this method, but still something that's nice about returning `PyObject`

Anyways, I didn't include such a change in this PR, so I'll just add the clippy annotation!

File: src/vocab.rs [hunk 3]
```diff
@@ -35,22 +35,27 @@ impl PyVocab {
         })
     }
 
-    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, usize)>>> {
+    #[allow(clippy::type_complexity)]
+    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, Option<usize>)>>> {
```
Comment (danieldk) [inline src/vocab.rs]: Oh, I didn't realize that the tuple became more complex. Maybe you could add a type alias

```
type NGramIndex = (String, Option<usize>);
```

?

Comment (sebpuetz) [reply src/vocab.rs]: Not too big of a fan of using aliases like this because I usually end up looking for its definition anyways. Tho, I pushed the change.

**Review by danieldk — APPROVED** [review]
